### DISCUSSION TITLE
How to pull to a local branch from remote origin main

### DISCUSSION THREAD

**Discussion by aidanmatchette** [jesseduffield/lazygit#1903] [Q&A]
How can I specify I want to pull from remote origin main into my local dev branch.

**Comment by Nauxscript** [comment]
I have the same demand; I always in the local dev branch coding and in order to get the new feature my colleague merge I need to pull origin master branch  into my local dev branch.

**Reply by mark2185** [reply]
Is there a specific reason why pulling/fetching `master` and then rebasing the local `dev` branch onto it is not an option?

**Reply by Nauxscript** [reply]
just for pulling origin master code and merge to local dev branch quickly and my leader not recommended to use rebase in case someone not up on git makes a mess

**Reply by mark2185** [reply]
Alright, so you don't `rebase`, but you `merge` when doing that? There's `M` for `merge into currently checked out branch`.

In any case, you can always run a custom command with by pressing `:` and typing in whatever you would normally do on the command line, it'll appear in the suggestions tab next time and you can just press `<Tab>` to select it.

**Reply by Nauxscript** [reply]
thank for your answer, maybe `:` will be useful.

**Comment by ooloth** [comment]
The way I do this is I...

1. Move to the Branches pane
2. Put my cursor on my local `main` branch
3. Press `f` to fetch the latest changes from `origin/main`
4. Press `M` to merge the updated local `main` branch into my checked-out local feature branch (cursor is still on `main`)

**Reply by javierheisecke-consultr** [reply]
Comment for future self. `M` = `shift+m`

**Comment by canton7** [comment]
Yes please!

From the branches pane, if you use `r` you get the option "Rebase on the base branch (origin/main)". It would be awesome to have the mirror option "Merge base branch (origin/main)" if you use `M`.
